pkg/capabilities/capabilities.go

```diff
@@ -105,6 +105,28 @@ type CallbackExecutable interface {
 	Execute(ctx context.Context, callback chan<- CapabilityResponse, request CapabilityRequest) error
 }
 
+type Validatable interface {
+	// GetConfigJsonSchema will create a values-wrapped JSON schema for the "config" field of the capability.
+	//
+	// This allows for more comprehensive validation of a workflow yaml file since
+	// after dynamically resolving the capability, the workflow engine can validate
+	// the config field against the schema.
+	//
+	// NOTE: What to do with method calls like:
+	//
+	//  - `RegisterToWorkflow` and `UnregisterFromWorkflow` 
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJsonSchema() (*CapabilityResponse)
```

Comment:
I expected
```suggestion
	ValidateConfig() (any, error)
```

Not sure if any can be made into a generic that returns the capability-specific interface.

Reply:
I was thinking of exposing another endpoint called `ValidateConfig` that returns nothing or a list of errors associated with the config.